[wasm][filesystem] add arg to testharness for setting working dir

tqiu8 opened the pull request:
fix #39998

tqiu8 committed:
[wasm][filesystem] add commandline arg to testharness for setting working dir

radical commented on src/mono/wasm/runtime-test.js:
```diff
@@ -119,6 +119,7 @@ setenv = {};
 runtime_args = [];
 enable_gc = true;
 enable_zoneinfo = false;
+working_dir = "net5.0-Browser-debug";
```
Instead of hardcoding this here, we should pass this in as an argument to `--working-dir` that you are adding.

radical commented on src/mono/wasm/runtime-test.js:
```diff
@@ -191,6 +195,8 @@ var Module = {
 		}
 
 		config.loaded_cb = function () {
+			FS.mkdir(working_dir);
+			FS.chdir(working_dir);
```
This seems to be breaking some tests. Eg:
` System.IO.FileNotFoundException : Could not load file or assembly '/net5.0-Browser-debug/System.Runtime.Loader.Tests.dll' or one of its dependencies.` (from https://github.com/dotnet/runtime/pull/42020/checks?check_run_id=1092348599)

I would suggest checking what is the current working directory for tests. You can print that in this method to check too. Then it should give a better idea of what should be passed in as the `--working-dir`.

radical committed:
[wasm][tests] Fix failing System.IO.Tests.DirectoryInfo_Name.CurrentDirectory

- This test was failing because it checks for
`Path.GetFileName(Directory.GetCurrentDirectory())` which returns `""`
for cwd==`"/"`.

- So, we add a dummy directory to the output, and use a new
`--working-dir` argument to set that as the cwd.

- And this is passed to the runtests script via a new msbuild property
called `$(RunTestsJSArguments)`

radical committed:
fix bad merge

lewing commented on src/mono/wasm/runtime-test.js:
```diff
@@ -119,6 +119,7 @@ setenv = {};
 runtime_args = [];
 enable_gc = true;
 enable_zoneinfo = false;
+working_dir='/';
```
what is the default working_dir generall?

radical replied:
I tried printing cwd for other tests, and it was `/`.

radical replied:
I mean, we weren't setting it at all. And `FS.cwd` said `/`

lewing reviewed (approved):
LGTM, thanks for taking care of this @tqiu8
